### PR TITLE
Make Teaser blocks dynamic

### PR DESCRIPTION
**Pull request by pbauer**
PLIP: https://github.com/plone/volto/issues/6023
Depends on https://github.com/plone/volto/pull/6024 but should not break anything without it.

### CODE REVIEW

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/serializer/blocks.py [hunk 1]
```diff
@@ -193,3 +198,82 @@ class SlateTableBlockSerializer(SlateTableBlockSerializerBase):
...
+@adapter(IPloneSiteRoot, IBrowserRequest)
+class TeaserBlockSerializerRoot(TeaserBlockSerializerBase):
+    """Serializer for site root"""
+
+
+def url_to_brain(url):
+    if not url:
+        return
+    brain = None
+    match = re.search("resolveuid/([^/]+)", url)
```
Comment (jensens) [inline src/plone/restapi/serializer/blocks.py:268]: Hint: a global `re.compile` before save CPU-cycles.

**Comment by sneridagh** [comment]
@pbauer could you take care of Jens suggestion?
@davisagli could you please take a look?

**Comment by pbauer** [comment]
@jenkins-plone-org please run jobs

**Review by jensens — APPROVED** [review]

**Review by davisagli — CHANGES_REQUESTED** [review]
This needs tests. I have some in a project that I might be able to adapt.

File: src/plone/restapi/serializer/blocks.py [hunk 2]
```diff
@@ -193,3 +198,84 @@ class SlateTableBlockSerializer(SlateTableBlockSerializerBase):
 @adapter(IPloneSiteRoot, IBrowserRequest)
 class SlateTableBlockSerializerRoot(SlateTableBlockSerializerBase):
     """Serializer for site root"""
+
+
+class TeaserBlockSerializerBase:
+    order = 0
```
Comment (davisagli) [inline src/plone/restapi/serializer/blocks.py:204]: Note to self: test whether this also works for teasers inside a grid. The resolveuid transform at the top level might happen too early and resolve the URLs in nested blocks before this runs there.

File: news/1788.feature [hunk 3]
```diff
@@ -0,0 +1 @@
+Make Teaser blocks dynamic @pbauer
```
Comment (davisagli) [inline news/1788.feature:1]:
```suggestion
Added TeaserBlockSerializer which updates the contents of a teaser block from its target if the block has `"overwrite": false`. @pbauer, @davisagli
```

**Review by davisagli — APPROVED** [review]

**Comment by davisagli** [comment]
@jenkins-plone-org please run jobs

**Comment by tisto** [comment]
@pbauer @davisagli good work! Shall we make a release with this new feature?

I am wondering if a major release would make it easier for us to keep things in sync with the Volto 18 release which this feature will be shipped with, right @sneridagh?

**Comment by sneridagh** [comment]
Yes, the plan is to have it paired with 18 and 6.1. I'm ok with a major, up to you though. 

We need a release asap, so we can have the Volto side merged. I plan to work on it during the sprint.

**Comment by tisto** [comment]
I gave this some thought and I will release a feature release. For "internal" packages it makes lots of sense to stick with strict semantic versioning.

**Comment by tisto** [comment]
@pbauer @davisagli @sneridagh FYI: I released this with p.restapi 9.7.0: https://pypi.org/project/plone.restapi/9.7.0/

**Pull request merged**